### PR TITLE
Osa cluster admin

### PR DESCRIPTION
**Pull request by JackQuincy**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_help.py [hunk 1]
```diff
@@ -648,11 +648,16 @@
   - name: --subnet-prefix
     type: string
     short-summary: The CIDR used on the Subnet into which to deploy the cluster.
+  - name: --customer-admin-group-id
+    type: string
+    short-summary: The ID of an Azure Active Directory Group that memberships will get synced into the OpenShift group "osa-customer-admins". If not specified no cluster admin access will be granted
```
Comment (mboersma) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_help.py]: Could you change the second sentence here to have a comma and end with a period?
`If not specified, no cluster admin access will be granted.`

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py [hunk 2]
```diff
@@ -2499,7 +2507,7 @@ def openshift_create(cmd, client, resource_group_name, name,  # pylint: disable=
 
     agent_infra_pool_profile = OpenShiftManagedClusterAgentPoolProfile(
         name='infra',  # Must be 12 chars or less before ACS RP adds to it
-        count=int(2),
+        count=int(3),
```
Comment (mboersma) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: Does this change the default node pool count from 2 to 3? That seems like a significant change (not related to this PR). Maybe it should be mentioned separately in the release notes.

Comment (julienstroheker) [reply src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: Yes, we agreed on this to switch from 2 to 3

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_help.py [hunk 3]
```diff
@@ -662,11 +662,16 @@
...
+  - name: --customer-admin-group-id
+    type: string
+    short-summary: The ID of an Azure Active Directory Group that memberships will get synced into the OpenShift group "osa-customer-admins". If not specified, no cluster admin access will be granted.
 
 
 examples:
   - name: Create an OpenShift cluster and auto create an AAD Client
     text: az openshift create -g MyResourceGroup -n MyManagedCluster --fqdn {FQDN}
+  - name: Create an OpenShift cluster and auto create an AAD Client and setup cluster admin group
+    text: az openshift create -g MyResourceGroup -n MyManagedCluster --fqdn {FQDN} --customer-admin-group-id {GROUP_ID}
```
Comment (tjprescott) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_help.py]: This is going to be incompatible with PR #9083 which removes FQDN.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py [hunk 4]
```diff
@@ -2253,9 +2254,14 @@ def _ensure_osa_aad(cli_ctx,
         # Delegate Sign In and Read User Profile permissions on Windows Azure Active Directory API
         resource_access = ResourceAccess(id="311a71cc-e848-46a1-bdf8-97ff7156d8e6",
                                          additional_properties=None, type="Scope")
-        required_osa_aad_access = RequiredResourceAccess(resource_access=[resource_access],
+        # Read directory permissions on Windows Azure Active Directory API
+        directory_access = ResourceAccess(id="5778995a-e1bf-45b8-affa-663a9f3f4d04",
```
Comment (tjprescott) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: What is this hard-coded ID?

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_help.py (hunk 3 above)
Comment (amanohar) [reply src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_help.py]: @tjprescott FQDN is not being removed. Its being made optional. PR #9083 will be done on top of this PR.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py (hunk 4 above)
Comment (amanohar) [reply src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: > What is this hard-coded ID?

@tjprescott these are scope GUIDs for programmatic use and represent various permissions: https://docs.microsoft.com/en-us/graph/permissions-reference. Currently there isn't a straightforward documentation from AAD for this and this obtained by looking at currently configured apps in AAD.

File: src/command_modules/azure-cli-acs/HISTORY.rst [hunk 5]
```diff
@@ -10,6 +10,10 @@ Release History
 ++++++
 * Minor fixes.
 
+2.3.20
+++++++
+* adding customer-admin-group-id flag to az osa create
```
[the remark this answers is not in the document]
Comment (sozercan) [reply src/command_modules/azure-cli-acs/HISTORY.rst]: thanks, moved to top as 2.3.22 and bumped version

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_help.py [hunk 6]
```diff
@@ -662,11 +662,16 @@
   - name: --subnet-prefix
     type: string
     short-summary: The CIDR used on the Subnet into which to deploy the cluster.
+  - name: --customer-admin-group-id
+    type: string
+    short-summary: The ID of an Azure Active Directory Group that memberships will get synced into the OpenShift group "osa-customer-admins". If not specified, no cluster admin access will be granted.
```
[the remark this answers is not in the document]
Comment (sozercan) [reply src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_help.py]: updated description

**Comment by amanohar** [comment]
@yugangw-msft @mboersma comments addressed by @sozercan. PTAL

**Comment by tjprescott** [comment]
@amanohar @sozercan need to get the CI green

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_help.py (hunk 1 above)
Comment (sozercan) [reply src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_help.py]: looks like this was already updated

**Comment by mboersma** [comment]
From the CI failures, maybe we just need to re-record the `test_aks_create_service_no_wait` test?

**Comment by sozercan** [comment]
@tjprescott any ideas why `PURPOSE='Load extension commands'` test fails? https://travis-ci.org/Azure/azure-cli/jobs/523726845

**Comment by sozercan** [comment]
@tjprescott @yugangw-msft tests are green now

**Pull request merged**